### ISSUE TITLE
CSS issue with the sticky cart submit container when the body tag doesn't have a defined background-color

### ISSUE THREAD

**Issue by webaxones** [woothemes/woocommerce#42286]
## Describe the bug

The `body` of a theme doesn't necessarily have a defined `background-color` value. However, the `wc-block-cart__submit-container--sticky` [relies on this value to define its own background color.](https://github.com/woocommerce/woocommerce-blocks/blob/7b8a191d893096bbaa4d99f0ac653a4f96038d8a/assets/js/blocks/cart/inner-blocks/proceed-to-checkout-block/block.tsx#L100)

So, in the case where the body has no defined background color, the result is as follows:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/17084006/910a8224-3021-4e15-ba2a-61309decf28e)

```HTML
<div class="wc-block-cart__submit-container wc-block-cart__submit-container--sticky" style="background-color: rgba(0, 0, 0, 0);">...
```

In the case where the body has the color `white` defined, the result is as follows:

![image](https://github.com/woocommerce/woocommerce-blocks/assets/17084006/49a0f640-a41f-4534-957b-125d56273d39)

```HTML
<div class="wc-block-cart__submit-container wc-block-cart__submit-container--sticky" style="background-color: rgb(255, 255, 255);">...
```

Even if it is discouraged for accessibility reasons not to define a background-color on the body tag, I think that `backgroundColor` should have a default value when the body tag does not.

## To reproduce

1. Make sure that the body tag of the active theme does not have a defined background-color, and if it does, remove this property
2. Test the cart block <780px
3. Same with a background-color on body and compare results

-   WordPress version: 6.2.2
-   WooCommerce version: 7.9.0
-   WooCommerce Blocks version: 10.6.1

### Desktop:

-   OS: macOS
-   Chrome

**Comment by Kallyan01** [comment]
Hello @webaxones ,
I was able to reproduce the bug and would like to work on fixing it. Could you please assign this issue to me?

Below is my proposed code change:

```js
const backgroundColor = useMemo( () => {
		const computedColor = getComputedStyle( document.body ).backgroundColor;
		if (
			! computedColor ||
			computedColor === 'rgba(0, 0, 0, 0)' ||
			computedColor === 'transparent'
		) {
			return '#fff'; // default fallback
		}

		return computedColor;
	}, [] );
```

This update ensures that if the body background is either undefined or fully transparent, a default background color of #fff is used as a fallback.

Let me know if this looks good to proceed.

**Comment by Kallyan01** [comment]
I have raised the fix PR . Let me know your thought on this . Thanks !

cc @kloon @nerrad @ralucaStan ,
